apps/frontend/src/modules/auth/slices/auth.slice.ts

```diff
@@ -38,6 +39,18 @@ const { actions, name, reducer } = createSlice({
 			state.dataStatus = DataStatus.REJECTED;
 		});
 
+		builder.addCase(uploadAvatar.pending, (state) => {
+			state.dataStatus = DataStatus.PENDING;
+		});
+
+		builder.addCase(uploadAvatar.fulfilled, (state, action) => {
+			if (state.authenticatedUser) {
+				state.authenticatedUser.avatarUrl = action.payload.data.url || null;
+			}
```

Comment:
Checking if the auth user exists in state is redundant  
```suggestion
		state.authenticatedUser = { ...state.authenticatedUser, avatarUrl: action.payload.data.url }
```